sandbox/lua/encoders/schema_influx.lua

```diff
@@ -176,13 +170,85 @@ function process_message()
         series = string.gsub(series_orig, "%%{([%w%p]-)}", sub_func)
     end
 
-    local output = {
+    output = {
         {
             name = series,
             columns = columns,
             points =  {values}
         }
     }
+
+    return output
+end
+
+local function process_multi_series(output, columns, values, place)
+    local output_index = 1
+    while true do
+        -- Iterate through Fields array in the message
+        local typ, name, value, representation, count = read_next_field()
+        -- Exit the perpetual loop when the iteration is complete
+        if not typ then break end
+
+        -- Only process fields that are not requested to be skipped
+        if not skip_fields_str or not skip_fields[name] then
+            local field_name = ""
+            local field_columns = {}
+            local field_points = {}
+
+            -- Instantiate a table in the output table for this iteration
+            output[output_index] = {}
+
+            -- Set the name attribute of this table appropriately
+            if use_subs then
+                series = string.gsub(series_orig, "%%{([%w%p]-)}", sub_func).."."..name
```

Comment:
The `string.gsub` call will return the same result each time, in a single `process_message` call. It should happen outside the fields loop.